src/win/conpty.cc

```diff
@@ -107,6 +135,8 @@ void SetupExitCallback(Napi::Env env, Napi::Function cb, pty_baton* baton) {
     // so we only call CloseHandle for now.
     CloseHandle(baton->hIn);
     CloseHandle(baton->hOut);
+    CloseHandle(baton->hShell);
+    assert(remove_pty_baton(baton->id));
```

Comment:
We were previously closing the client handle in `pty.kill` which would invalidate waiting on the exit callback. Moved it closer to when the client has signaled exit.